src/HttpClientFactory/Http/test/Logging/HttpHeadersLogValueTest.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+using Microsoft.Extensions.Http.Logging;
+using Xunit;
+
+namespace Microsoft.Extensions.Http.Tests.Logging
+{
+    public class HttpHeadersLogValueTest
+    {
+        [Fact]
+        public void HttpHeadersLogValue_ToString_HidesOnlyLogSensitiveHeadersValue()
+        {
+            // Arrange
+            var headers = new TestHttpHeaders
+            {
+                { "secureHeader1", "value1" },
+                { "unsecureHeader1", "value1" }
+            };
+            var contentHeaders = new TestHttpHeaders
+            {
+                { "unsecureHeader2", "value2" },
+                { "secureHeader2", "value2" }
+            };
+            var logSensitiveHeaders = new HashSet<string>
+            {
+                "secureHeader1",
+                "secureHeader2",
+            };
+            var sensitiveHeaders = new HashSet<string>(logSensitiveHeaders, StringComparer.OrdinalIgnoreCase);
+
+            Predicate<string> isSensitiveHeader = (header) => sensitiveHeaders.Contains(header);
+
+            var httpHeadersLogValue = new HttpHeadersLogValue(HttpHeadersLogValue.Kind.Request, headers, contentHeaders, isSensitiveHeader);
+
+            // Act
+            var result = httpHeadersLogValue.ToString();
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(
+                "Request Headers:" + Environment.NewLine +
+                "secureHeader1: *" + Environment.NewLine +
+                "unsecureHeader1: value1" + Environment.NewLine +
+                "unsecureHeader2: value2" + Environment.NewLine +
+                "secureHeader2: *" + Environment.NewLine,
+                result);
+        }
+
+        private class TestHttpHeaders : HttpHeaders { }
+    }
```

Comment:
I'd also like to see an integration test or two - you can use a patterrn like [this](https://github.com/aspnet/Extensions/blob/49bf9920e4e9079b02e2f775b2bc923abf2be4d1/src/HealthChecks/HealthChecks/test/HealthCheckPublisherHostedServiceTest.cs#L165) to hijack logging for testing.

Reply:
@a-tsymbal are you able to add an integration test? I can help with this if you get stuck

Reply:
@rynowak It would be great if you can help a little bit.